### PR TITLE
Add Float TLV

### PR DESCRIPTION
**Pull request by mfucci**

### CODE REVIEW

File: src/tlv/TlvFloat.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * Copyright 2022 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DataReaderLE } from "../util/DataReaderLE";
+import { DataWriterLE } from "../util/DataWriterLE";
+import { TlvType, TlvCodec, TlvTag } from "./TlvCodec";
+import { TlvSchema } from "./TlvSchema";
+
+const FLOAT_RANGE = { min: -340282346638528859811704183484516925440.0, max: 340282346638528859811704183484516925440.0 };
```
Comment (turon) [inline src/tlv/TlvFloat.ts:12]: Perhaps formally define FLOAT_MAX and FLOAT_MIN and use those to define FLOAT_RANGE?
Seems like FLOAT_MAX and FLOAT_MIN as separate consts may come in handy later.

Comment (mfucci) [reply src/tlv/TlvFloat.ts]: Done

File: test/tlv/TlvBooleanTest.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                assert.strictEqual(arrayBufferToHex(TlvBoolean.encode(input)), out);
+            });
+        }
+    });
+
+    context("decode", () => {
+        for (const testName in decodeTestVector) {
+            const { input, out } = decodeTestVector[testName];
+            it(testName, () => {
+                assert.strictEqual(TlvBoolean.decode(arrayBufferFromHex(input)), out);
```
Comment (turon) [inline test/tlv/TlvBooleanTest.ts:38]: Can't we just use one test vector and reverse `in` / `out` for `encode` / `decode`?

File: src/tlv/TlvFloat.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    /** @override */
+    validate(value: number): void {
+        if (this.min !== undefined && value < this.min) throw new Error(`Invalid value: ${value} is below the minimum, ${this.min}.`);
+        if (this.max !== undefined && value > this.max) throw new Error(`Invalid value: ${value} is above the maximum, ${this.min}.`);
+    }
+}
+
+
+/** Double TLV schema bounded by a min / max. */
+export const TlvBoundedDouble = ({ min, max }: { min?: number, max?: number } = {}, type?: TlvType.Float32 | TlvType.Float64) => new FloatSchema(min, max, type);
```
Comment (turon) [inline src/tlv/TlvFloat.ts:53]: Could this be refactored for readability?  It's impressively compact, but difficult to understand in its current form.

File: src/util/DataWriterLE.ts [hunk 4]
```diff
@@ -4,34 +4,84 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+function toNumber(value: bigint | number): number {
```
Comment (turon) [inline src/util/DataWriterLE.ts:7]: Can these be static under some utility class or namespace to simplify imports?

File: test/tlv/TlvBooleanTest.ts (hunk 2 above)
Comment (Apollon77) [reply test/tlv/TlvBooleanTest.ts:38]: Aahh then I missed the separation ...sorry

**Review by turon — APPROVED** [review]
My comments are non-blocking.  Looks good!

File: test/tlv/TlvBooleanTest.ts (hunk 2 above)
Comment (mfucci) [reply test/tlv/TlvBooleanTest.ts:38]: As discussed offline, I will update that in PR #20

File: src/util/DataWriterLE.ts (hunk 4 above)
Comment (mfucci) [reply src/util/DataWriterLE.ts:7]: Already done in a few PRs later in the chain :-) (I created Number.ts that contains min / max and those functions

**Pull request merged**